site/docs/expressions/extended_expression.md

```diff
@@ -1,19 +1,25 @@
 
-Besides, as it will be used seperately with plan rel representation, it need include basic fields like Version.
+For details, see the definition in [extended_expression.proto](https://github.com/substrait-io/substrait/blob/main/proto/substrait/extended_expression.proto).
+
+## Referred expression
+
+Extended expressions can contain one or multiple expressions. A single expression is used for filters, while multiple expressions are used for projections and aggregations. Each of these expressions is represented by an `ExpressionReference` message, which can either contain an [Expression](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto) or [AggregateFunction](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto#L1170) message. In the future, other types might be allowed to handle different use cases.
 
 ## Input and output data schema
 
-Similar as `base_schema` defined in [ReadRel](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto#L58), the input data schema tells name/type/nullibilty and layout info of input data for target expression evalutation. It also has a field `name` to define name of output data.
+The schema of the input data is specified by the `base_schema` field.
 
-## Referred expression
+The schema of the output data is determined by each of the `ExpressionReference` messages in the `referred_expr` field. The output type for the expression is determined by the expression or measure. The field name is determined by `output_names`. For primitive types `output_names` is just one string, but if the output type is a struct it will contain a name for each field, in depth-first order.
 
-It will has one or more referred expressions in this message and the referred expressions can be [Expression](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto) or [AggregateFunction](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto#L1170). More types of expression can be added for more scenarios.
+## Order
 
-For multi expressions, user can translate them following same order as it occurs in original plan rel. But it does NOT require the consume side to handle it strictly in previous order. Only need to make sure columns in final output are organized in same order as defined in extended expression message.
+If there are multiple expressions, the order of the inputs and outputs matter. Input data will match the schema order
```

Comment:
What does "the order of the inputs matter" mean?

I agree that the output schema must align with the expression order.

Reply:
Changed to "the field order of the input and output data matters".